fuel/datasets/base.py

```diff
@@ -28,10 +29,6 @@ class Dataset(object):
         'targets')`` for MNIST (regardless of which data the data stream
         actually requests). Any implementation of a dataset should set this
         attribute on the class (or at least before calling ``super``).
-    default_iteration_scheme : :class:`.IterationScheme`, optional
```

Comment:
> However, that default would crash our most basic tutorial simply because most people use minibatches and expect 2D matrices, which would make the current `SequentialScheme` a better default. 

Do our tutorial use default scheme? At least http://blocks.readthedocs.org/en/latest/tutorial.html does not. And there is no sense to have a tutorial without showing how to specify the batch size.

Counter-proposal: instead of `get_default_scheme` we provide `get_example_stream` which provides stream of examples. This is clearer than default, and this is the only reasonable default, sorry for this poor word-play. I just do not like that currently I can not just iterate over a given dataset, I have to check its type and create a data stream with a custom scheme.